src/basic/selinux-util.c

```diff
@@ -66,32 +69,34 @@ bool mac_selinux_use(void) {
 #endif
 }
 
+bool mac_selinux_enforcing(void) {
+#if HAVE_SELINUX
+        return enforcing_status_func() != 0;
+#else
+        return false;
+#endif
+}
+
 void mac_selinux_retest(void) {
 #if HAVE_SELINUX
         cached_use = -1;
 #endif
 }
 
-int mac_selinux_init(void) {
 #if HAVE_SELINUX
+static int open_label_db(struct selabel_handle **hnd) {
         usec_t before_timestamp, after_timestamp;
         struct mallinfo before_mallinfo, after_mallinfo;
         char timespan[FORMAT_TIMESPAN_MAX];
         int l;
 
-        selinux_set_callback(SELINUX_CB_POLICYLOAD, (union selinux_callback) mac_selinux_reload);
-
-        if (label_hnd)
-                return 0;
-
-        if (!mac_selinux_use())
-                return 0;
+        assert(hnd);
 
         before_mallinfo = mallinfo();
         before_timestamp = now(CLOCK_MONOTONIC);
 
-        label_hnd = selabel_open(SELABEL_CTX_FILE, NULL, 0);
-        if (!label_hnd)
+        *hnd = selabel_open(SELABEL_CTX_FILE, NULL, 0);
+        if (!*hnd)
```

Comment:
nah, please do not touch return parameters on failure.

we generally have the rule, that return parameters should always be left untouched on failure, and always be initialized on success (see CODING_STYLE). i.e. in this case assign the result of selabel_open() to an internal variable first, and when everything is good, assign it to the return parameter.

Reply:
refactored `open_label_db()`